### PR TITLE
2024-04-interlis-export-orientation-option

### PR DESCRIPTION
**Pull request by sjib**

### CODE REVIEW

**Comment by sjib** [comment]
Should solve https://github.com/teksi/wastewater/issues/216

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py [hunk 1]
```diff
@@ -29,6 +29,12 @@ def __init__(self, parent):
             [config.MODEL_NAME_VSA_KEK, config.MODEL_NAME_SIA405_ABWASSER],
         )
 
+        # Fill orientation selection combobox
+        self.export_orientation_selection_comboBox.clear()
+        self.export_orientation_selection_comboBox.addItem("90.0", "+90°")
+        self.export_orientation_selection_comboBox.addItem("0.0", "0°")
+        self.export_orientation_selection_comboBox.addItem("-90.0", "-90°")
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py:36]:
```suggestion
        self.export_orientation_selection_comboBox.addItem("90°", 90.0)
        self.export_orientation_selection_comboBox.addItem("0°", 0.0)
        self.export_orientation_selection_comboBox.addItem("-90°", -90.0)
```

Here you could add data directly as number so you don't need to convert it back after

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py [hunk 2]
```diff
@@ -106,3 +112,10 @@ def selected_model(self):
 
     def selected_models(self):
         return self.export_model_selection_comboBox.currentData()
+
+    def selected_orientation(self):
+        # neu 23.4.2024 added float()
+        eorientation = float(self.export_orientation_selection_comboBox.currentText())
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py:118]:
```suggestion
        eorientation = self.export_orientation_selection_comboBox.currentData()
```

And than you can use [currentData](https://doc.qt.io/qt-5/qcombobox.html#currentData-prop) at this place

File: plugin/teksi_wastewater/interlis/gui/interlis_importer_exporter_gui.py [hunk 3]
```diff
@@ -113,6 +113,8 @@ def action_export(self):
                 export_models=export_dialog.selected_models(),
                 logs_next_to_file=export_dialog.logs_next_to_file,
                 limit_to_selection=export_dialog.limit_to_selection,
+                # export orientation added
+                export_orientation=export_dialog.selected_orientation,
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_importer_exporter_gui.py:117]:
```suggestion
                export_orientation=export_dialog.selected_orientation(),
```

Or you could add the `@property` decoration like for logs_next_to_file, limit_to_selection, etc...

Comment (sjib) [reply plugin/teksi_wastewater/interlis/gui/interlis_importer_exporter_gui.py:117]: @domi4484 I will go for  `@property` - thanks for all the hints.

File: plugin/teksi_wastewater/interlis/interlis_model_mapping/interlis_exporter_to_intermediate_schema.py [hunk 4]
```diff
@@ -2390,12 +2390,16 @@ def truncate(self, val, max_length):
...
-    def _modulo_angle(self, val):
+    def _modulo_angle(self, val, selected_orientation):
         """
         Returns an angle between 0 and 359.9 (for Orientierung in Base_d-20181005.ili)
         """
         if val is None:
             return None
+
+        # add selected_orientation
+        val = val + selected_orientation
```
Comment (sjib) [inline plugin/teksi_wastewater/interlis/interlis_model_mapping/interlis_exporter_to_intermediate_schema.py:2401]: @domi4484 Is this enough to add the selected_orientation to val or do I need to adapt something else in the function call?

https://github.com/teksi/wastewater/blob/94434d5e3293b148adb20c31b490bd0262f5f6d5/plugin/teksi_wastewater/interlis/interlis_model_mapping/interlis_exporter_to_intermediate_schema.py#L2627

Comment (domi4484) [reply plugin/teksi_wastewater/interlis/interlis_model_mapping/interlis_exporter_to_intermediate_schema.py:2401]: No you have to pass the angle to the modulo function and to the exporter

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py [hunk 5]
```diff
@@ -106,3 +112,10 @@ def selected_model(self):
 
     def selected_models(self):
         return self.export_model_selection_comboBox.currentData()
+
+    @property
+    def selected_orientation(self):
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py:117]:
```suggestion
    def labels_orientation_offset(self):
```
I would rename it label orientation offset or similar for more readability

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py [hunk 6]
```diff
@@ -106,3 +112,10 @@ def selected_model(self):
...
+        # neu 23.4.2024 added float()
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py:118]: Comment is outdated

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py [hunk 7]
```diff
@@ -106,3 +112,10 @@ def selected_model(self):
...
+        eorientation = self.export_orientation_selection_comboBox.currentData()
+        print(eorientation)
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py:120]: Print should be removed or replaced by a log

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.ui [hunk 8]
```diff
@@ -52,17 +52,35 @@
...
+       <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
-         <widget class="QLabel" name="labels_qgis_warning_label">
-          <property name="styleSheet">
-           <string notr="true">*{color: red; }</string>
-          </property>
+         <widget class="QLabel" name="orientation_label">
           <property name="text">
-           <string>This feature requires QGIS 3.26.2 or newer.</string>
+           <string>Angle for horizontal text in INTERLIS (default: 90.0°)</string>
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.ui:59]: In other part of the dialog we use the term "label" and here "text". Should we choose one of them for consistency?
ìn INTERLIS` can be removed as already in the title of the dialog?

File: plugin/teksi_wastewater/interlis/interlis_importer_exporter.py [hunk 9]
```diff
@@ -153,6 +153,8 @@ def interlis_export(
         export_models,
         logs_next_to_file=True,
         limit_to_selection=False,
+        # neu export_orientation
+        export_orientation=90,
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:157]:
```suggestion
        export_orientation=90.0,
```
So it is initialized as float

File: plugin/teksi_wastewater/interlis/interlis_importer_exporter.py [hunk 10]
```diff
@@ -182,6 +184,8 @@ def interlis_export(
             labels_file_path = os.path.join(tempdir.name, "labels.geojson")
             self._export_labels_file(
                 limit_to_selection=limit_to_selection,
+                # neu export orientation
+                export_orientation=export_orientation,
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:188]: Around line 195 `self._export_to_intermediate_schema(` also needs this parameter to pass it to `InterlisExporterToIntermediateSchema`

**Review by domi4484 — COMMENTED** [review]
All the comments `# neu export orientation` can be removed as they don't make the code more readable and are its clear by git that this option was added in a second time

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.ui (hunk 8 above)
Comment (sjib) [reply plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.ui]: OK adjusted

File: plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py (hunk 7 above)
Comment (sjib) [reply plugin/teksi_wastewater/interlis/gui/interlis_export_settings_dialog.py]: ok

File: plugin/teksi_wastewater/interlis/interlis_importer_exporter.py [hunk 11]
```diff
@@ -182,16 +184,19 @@ def interlis_export(
...
             )
 
         # Export to the temporary ili2pg model
-        self._progress_done(35, "Converting from Teksi Wastewater...")
+        self._progress_done(35, "Converting from TEKSI Wastewater...")
         self._export_to_intermediate_schema(
             export_model=export_models[0],
             file_name=xtf_file_output,
             selected_ids=selected_ids,
+            export_orientation=export_orientation,
```
Comment (sjib) [inline plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:199]: also added export_orientation=export_orientation

File: plugin/teksi_wastewater/interlis/interlis_importer_exporter.py (hunk 10 above)
Comment (sjib) [reply plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:187]: @domi4484 Is this better now?

File: plugin/teksi_wastewater/interlis/interlis_importer_exporter.py [hunk 12]
```diff
@@ -341,6 +350,7 @@ def _export_to_intermediate_schema(
         export_model,
         file_name=None,
         selected_ids=None,
+        export_orientation=90.0,
```
Comment (domi4484) [inline plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:353]: Here you also need to pass this parameter to `InterlisExporterToIntermediateSchema`

Comment (sjib) [reply plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:353]: `export_orientation=selected_orientation`

Comment (sjib) [reply plugin/teksi_wastewater/interlis/interlis_importer_exporter.py:353]: Seems not to work like that:
`==================================== ERRORS ====================================
___________ ERROR collecting teksi_wastewater/tests/test_interlis.py ___________
teksi_wastewater/tests/test_interlis.py:8: in <module>
    from teksi_wastewater.interlis.interlis_importer_exporter import (
teksi_wastewater/interlis/interlis_importer_exporter.py:56: in <module>
    class InterlisImporterExporter:
teksi_wastewater/interlis/interlis_importer_exporter.py:156: in InterlisImporterExporter
    export_orientation=selected_orientation,
E   NameError: name 'selected_orientation' is not defined
=========================== short test summary info ============================
ERROR teksi_wastewater/tests/test_interlis.py - NameError: name 'selected_ori...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.73s ===============================
2`

**Review by domi4484 — APPROVED** [review]

**Pull request merged**